### PR TITLE
[Update] Deploy NodeBalancers with the Linode Cloud Controller Manager

### PR DESCRIPTION
**Pull request by leslitagordita**
New
* Configuring Load Balancing with TLS Encryption on a Kubernetes Cluster
* Getting Started with Load Balancing on a Linode Kubernetes Engine (LKE) Cluster
* A Tutorial for Installing the Linode CCM on an Unmanaged Kubernetes Cluster

Update
* Understanding Kubernetes cert-manager (used to be: A Tutorial for Deploying and Managing TLS Certificates on Kubernetes with Cert-Manager)
* Deploy NodeBalancers with the Linode Cloud Controller Manager (removed and aliases to "Getting Started with Load Balancing on a Linode Kubernetes Engine (LKE) Cluster)"

### CODE REVIEW

**Comment by nmelehan** [comment]
Deploy preview for *nostalgic-ptolemy-b01ab8* ready!

Built with commit 369f6b789440ad695f1af2f204cef57cede083cd

https://deploy-preview-3473--nostalgic-ptolemy-b01ab8.netlify.app

File: docs/kubernetes/installing-the-linode-ccm-on-an-unmanaged-kubernetes-cluster/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    {{< note >}}
+It is recommended that you install the Linode CCM on a new Kubernetes cluster, as there are a number of issues that prevent the CCM from running on Nodes that are in the "Ready" state.
+    {{</ note >}}
+
+1. Ensure you have [kubectl installed](/docs/applications/configuration-management/terraform/how-to-provision-an-unmanaged-kubernetes-cluster-using-terraform/#install-kubectl) on your local computer and you can access your Kubernetes cluster with it.
+
+1. [Install Git](/docs/development/version-control/how-to-install-git-on-linux-mac-and-windows/) on your local computer.
+
+1. Generate a [Linode APIv4 token](/docs/platform/api/getting-started-with-the-linode-api/#get-an-access-token)).
```
Comment (andystevensname) [inline docs/kubernetes/installing-the-linode-ccm-on-an-unmanaged-kubernetes-cluster/index.md:53]: Extra ) at the end of the sentence.

File: docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+{{< note >}}
+The [Linode Terraform K8s module](http://localhost:1313/docs/applications/configuration-management/terraform/how-to-provision-an-unmanaged-kubernetes-cluster-using-terraform/) also deploys a Kubernetes cluster with the Linode CCM installed by default. Any Kubernetes cluster with a Linode CCM installation can make use of Linode NodeBalancers in the ways described in this guide.
+{{</ note>}}
+
+## In this Guide
+
+This guide will show you:
+
+- manifest file configurations needed to [add Linode NodeBalancers to your LKE cluster](#adding-linode-nodebalancers-to-your-kubernetes-cluster).
+- [annotations](/#annotations-reference) available to further configure your Linode NodeBalancers behavior and how to incorporate them into a manifest file.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md]: fixed

File: docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+### Before You Begin
+
+This guide assumes you have a working Kubernetes cluster that was deployed using the Linode Kubernetes Engine (LKE). You can deploy a Kubernetes cluster using LKE in the following ways:
+
+- The [Linode Cloud Manager](/docs/kubernetes/deploy-and-manage-a-cluster-with-linode-kubernetes-engine-a-tutorial/).
+- [Linode's API v4](/docs/kubernetes/deploy-and-manage-lke-cluster-with-api-a-tutorial/).
+- [Terraform](/docs/kubernetes/how-to-deploy-an-lke-cluster-using-terraform/), the popular infrastructure as code (IaC) tool.
+
+    {{< note >}}
+An LKE cluster will already have Linode's Cloud Controller Manager installed in the cluster's control plane. If you **did not** deploy your Kubernetes cluster using LKE and would like to make use of the Linode Cloud Controller Manager, see How to Add NodeBalancers to a Non-Linode Managed Kubernetes Cluster.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md]: fixed

File: docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    {{</ note >}}
+
+
+## Adding Linode NodeBalancers to your Kubernetes Cluster
+
+To add an external load balancer to your Kubernetes cluster you can add the example lines to a new configuration file or more commonly, to a Service file. When the configuration is applied to your cluster, Linode NodeBalancers will be created, and added to your Kubernetes cluster. Your cluster will be accessible via a public IP address and the NodeBalancers will route external traffic to a Service running on healthy nodes in your cluster.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md]: fixed

File: docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+Events:                   <none>
+      {{</ output >}}
+
+
+## Configuring your Linode NodeBalancers with Annotations
+
+The Linode CCM accepts annotations that configure the behavior and settings of your cluster's underlying NodeBalancers.
+
+- The table below provides a list of all available annotation suffixes.
+- Each annotation **must** be prefixed with `service.beta.kubernetes.io/linode-loadbalancer-`. For example, the complete for the `throttle` annotation is `service.beta.kubernetes.io/linode-loadbalancer-throttle`.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md]: fixed

File: docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+{{< file "example-serivce.yaml" yaml >}}
+...
+metadata:
+  annotations:
+    service.beta.kubernetes.io/linode-loadbalancer-default-protocol: http
+    service.beta.kubernetes.io/linode-loadbalancer-port-443: '{ "tls-secret-name": "example-secret", "protocol": "https" }'
+...
+{{</ file >}}
+
+- The `service.beta.kubernetes.io/linode-loadbalancer-default-protocol` configures the NodeBalancer's default protocol to use.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md]: fixed

File: docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+
+To delete a NodeBalancer and the Service that it represents, you can use the Service manifest file you used to create the NodeBalancer. Simply use the `delete` command and supply your file name with the `f` flag:
+
+    kubectl delete -f example-service.yaml
+
+Similarly, you can delete the Service by name:
+
+    kubectl delete service example-service
+
+After deleting your service, its corresponding NodeBalancer will be removed from your Linode account.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/getting-started-with-load-balancing-on-a-lke-cluster/index.md]: Good call. I added a note about the `preserve` annotation and linked to the annotation reference for more details.

File: docs/kubernetes/how-to-configure-load-balancing-with-tls-encryption-on-a-kubernetes-cluster/index.md [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+  name: Linode
+title: "How to Configure Load Balancing with TLS Encryption on a Kubernetes Cluster"
+h1_title: "Configuring Load Balancing with TLS Encryption on a Kubernetes Cluster"
+contributor:
+  name: Linode
+external_resources:
+- '[NGINX Ingress Controller User Guide](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/)'
+---
+
+This guide will use an example Kubernetes Deployment and Service to demonstrate how to route external traffic to a Kubernetes application over HTTPS. This is accomplished using the [NGINX Ingress Controller](https://kubernetes.github.io/ingress-nginx/deploy/#using-helm), [cert-manager](https://cert-manager.io/docs/), and [Linode NodeBalancers](/docs/platform/nodebalancer/). The NGINX Ingress Controller, uses Linode NodeBalancers, which is Linode's load balancing service, to route a Kubernetes Service's traffic to the appropriate backend Pods over HTTP and HTTPS. cert-manager creates a Transport Layer Security (TLS) certificate from the [Let’s Encrypt](https://letsencrypt.org/) certificate authority (CA) providing secure HTTPS access to a Kubernetes Service.
```
[the remark this answers is not in the document]
Comment (leslitagordita) [reply docs/kubernetes/how-to-configure-load-balancing-with-tls-encryption-on-a-kubernetes-cluster/index.md]: fixed

File: docs/kubernetes/installing-the-linode-ccm-on-an-unmanaged-kubernetes-cluster/index.md (hunk 1 above)
Comment (leslitagordita) [reply docs/kubernetes/installing-the-linode-ccm-on-an-unmanaged-kubernetes-cluster/index.md]: fixed

**Pull request closed without merging**